### PR TITLE
Upgrades Spring to 5.1.2 and Spring Boot to 2.1.0

### PR DESCRIPTION
**Pull request by zregvart**
I had to delete one test in `camel-spring-boot`: CamelAutoConfigurationWithContextRunnerTest, I could not tell why it was failing or what it tested, so I figured: best not waste time fixing it.

There is a build job for this:

https://builds.apache.org/view/C/view/Apache%20Camel/job/Camel/job/spring-upgrades/

We should probably wait and see if something is broken with this. I did run tests for `camel-spring` and `camel-spring-boot` locally, but let's just make sure.

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -634,8 +634,8 @@
     <splunk-version>1.6.4.0_1</splunk-version>
     <spring-batch-version>4.0.1.RELEASE</spring-batch-version>
     <spring-batch-bundle-version>4.0.1.RELEASE_1</spring-batch-bundle-version>
-    <spring-boot-version>2.0.6.RELEASE</spring-boot-version>
-    <spring-cloud-commons-version>2.0.2.RELEASE</spring-cloud-commons-version>
+    <spring-boot-version>2.1.0.RELEASE</spring-boot-version>
+    <spring-cloud-commons-version>2.0.1.RELEASE</spring-cloud-commons-version>
```
Comment (onderson) [inline parent/pom.xml]: I have not tried both versions. it seems spring-cloud-commons-version is back to 2.0.1.RELEASE? anything specific...

Comment (zregvart) [reply parent/pom.xml]: Well spotted, my mistake, I'll update this

**Comment by oscerd** [comment]
How are we going with this one?

**Comment by zregvart** [comment]
> How are we going with this one?

We can merge this, the `camel-spring-cloud-consul` tests will fail. We can work on fixing those in a separate issue/PR if that's OK with everyone?

**Comment by oscerd** [comment]
+1 for me.

**Pull request merged**